geoviews/plotting/mpl/chart.py

```diff
@@ -0,0 +1,156 @@
+    information about the vector magnitude by putting slanted lines or a
+    triangle for various increments in magnitude.
+
+    The largest increment is given by a triangle (or "flag"). After those come full lines (barbs).
+    The smallest increment is a half line. There is only, of course, ever at most 1 half line.
+    If the magnitude is small and only needs a single half-line and no full lines or
+    triangles, the half-line is offset from the end of the barb so that it can be
+    easily distinguished from barbs with a single full line. The magnitude for the barb
+    shown above would nominally be 65, using the standard increments of 50, 10, and 5.
+    """
+
+    padding = param.ClassSelector(default=0.05, class_=(int, float, tuple))
+
+    from_uv_components = param.Boolean(default=False, doc="""
+        If True, the vdims are expected to be in the form of
+        wind components (U and V); else, wind direction and speed.""")
+
+    style_opts = [
+        "alpha",
+        "color",
+        "edgecolors",
+        "facecolors",
+        "linewidth",
+        "marker",
+        "visible",
+        "cmap",
+        "norm",
+        # barb specific
+        "length",
+        "barbcolor",
+        "flagcolor",
+        "fill_empty",
+        "rounding",
+        "barb_increments",
+        "flip_barb",
+        "pivot",
+        "sizes",
+        "width",
+    ]
+
+    _nonvectorized_styles = [
+        "alpha",
+        "marker",
+        "cmap",
+        "visible",
+        "norm",
+        # TODO: clarify whether these are vectorized or not
+        "length",
+        "barbcolor",
+        "flagcolor",
+        "fill_empty",
+        "rounding",
+        "barb_increments",
+        "flip_barb",
+        "pivot",
+        "sizes",
+        "width",
+    ]
+
+    _plot_methods = dict(single="barbs")
+
+    def _get_us_vs(self, element):
+        if self.from_uv_components:
+            us = element.dimension_values(2, flat=False) if len(element.data) else []
+            vs = element.dimension_values(3, flat=False) if len(element.data) else []
+            uv_magnitudes = np.hypot(us, vs)  # unscaled
+            radians = (np.pi / 2.0) - np.arctan2(us / uv_magnitudes, vs / uv_magnitudes)
+            element = element.add_dimension("Angle", 4, radians, vdim=True)
+            element = element.add_dimension("Magnitude", 5, uv_magnitudes, vdim=True)
+            mag_dim = element.get_dimension(5)
+        else:
+            radians = element.dimension_values(2) if len(element.data) else []
+            mag_dim = element.get_dimension(3)
+
+        if isinstance(mag_dim, str):
+            mag_dim = element.get_dimension(mag_dim)
+
+        if isinstance(mag_dim, dim):
+            magnitudes = mag_dim.apply(element, flat=True)
+        else:
+            magnitudes = element.dimension_values(mag_dim)
+
+        if self.invert_axes:
+            radians += 1.5 * np.pi
```

Comment:
This is just rotating the coordinates 270 degrees, correct? At the top of my head, I would think only 180 degrees are needed.

Reply:
I believe this is correct because x / y axes are flipped. We don't want to just flip the direction.

<img width="792" alt="image" src="https://github.com/holoviz/geoviews/assets/15331990/4dca22af-5b6c-43a7-829c-5975d6cbe521">

Reply:
As a sanity check: invert_axes just swaps the U and V args I think?
U, V
<img width="422" alt="image" src="https://github.com/holoviz/geoviews/assets/15331990/1da2037f-7fc7-4f75-ab7b-2c85d4238a95">


V, U
<img width="507" alt="image" src="https://github.com/holoviz/geoviews/assets/15331990/6de08303-d6ff-4aa0-8649-5ec790521380">

Reply:
I made it rotate 90 degrees now.

Reply:
I forgot that `invert_axes` swap the axes and does not invert both axes independently, so this sound right.